sigopt/experiment_context.py

```diff
@@ -0,0 +1,66 @@
+import threading
+
+from .interface import get_connection
+from .run_factory import BaseRunFactory
+
+
+class ExperimentContext(BaseRunFactory):
+  '''Wraps the Experiment object and provides extra utility methods.'''
+
+  def __init__(self, experiment):
+    if experiment.project is None:
+      raise ValueError("experiment does not belong to a project")
```

Comment:
`get_experiment` (which hasn't been implemented yet) would have this issue.

Moving the experiment does not create runs from existing observations, so this object is expected to be pretty useless if the goal was to fetch data in the experiment.

Creating an experiment on the web and then using that experiment in the client is not commonly done(?) and should be scoped for the project pages if we expect it to be a common user story

Reply:
I thought experiments were always added to projects now, or that was the goal at least - so I figured this would be a "deprecation" type message we could get rid of after a bit? Or is that not how it's going to work?

Reply:
All experiments created with this client will belong to a project, so this error is more of an assertion than anything. I can't think of a catch-all recommendation for the user to do if the experiment does not belong to a project.

Reply:
Ah gotcha. Ok, that's fine as-is then, and thanks for explaining